Userland/Libraries/LibWeb/DOM/RadioNodeList.h

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (c) 2023, Shannon Booth <shannon@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <LibWeb/DOM/NodeList.h>
+
+namespace Web::DOM {
+
+// https://html.spec.whatwg.org/multipage/common-dom-interfaces.html#radionodelist
+class RadioNodeList : public NodeList {
+    WEB_PLATFORM_OBJECT(RadioNodeList, NodeList);
+
+public:
+    [[nodiscard]] static JS::NonnullGCPtr<RadioNodeList> create(JS::Realm&, Vector<JS::Handle<Element>> const&);
+
+    virtual ~RadioNodeList() override;
+
+    virtual u32 length() const override;
+    virtual Node const* item(u32) const override;
+
+protected:
+    virtual void initialize(JS::Realm&) override;
+
+private:
+    explicit RadioNodeList(JS::Realm&, Vector<JS::Handle<Element>> const&);
+
+    Vector<JS::Handle<Element>> m_node_list;
```

Comment:
Not 100% on whether this should be an `Element` or not. Also using a Handle right now, but maybe this should be GCCounted? Really unsure of the differences between all that

Reply:
As I understand it, `RadioNodeList` is supposed to be a "live" collection, meaning that if radio elements are removed from the DOM, they should no longer be present in the `RadioNodeList` either. So I think we shouldn't be storing them here like this in the first place :)

Reply:
Right, that would make a lot more sense, and explains my difficulty! Glad that doing the wrong thing was difficult :P

Will give this another shot, thanks!

Reply:
Thanks, this looks much better now!